src/main/java/org/opensearch/knn/index/codec/KNN920Codec/KNN920Codec.java

```diff
@@ -49,4 +65,63 @@ public DocValuesFormat docValuesFormat() {
     public CompoundFormat compoundFormat() {
         return knnFormatFacade.compoundFormat();
     }
+
+    @Override
+    public KnnVectorsFormat knnVectorsFormat() {
+        return perFieldKnnVectorsFormat;
+    }
+
+    /**
+     * Class provides per field format implementation for Lucene Knn vector type
+     */
+    class KNN920PerFieldKnnVectorsFormat extends PerFieldKnnVectorsFormat {
+
+        @Override
+        public KnnVectorsFormat getKnnVectorsFormatForField(final String field) {
+            if (isNotKnnVectorFieldType(field)) {
+                log.debug(
+                    String.format(
+                        "Initialize KNN vector format for field [%s] with default params [max_connections] = \"%d\" and [beam_width] = \"%d\"",
+                        field,
+                        Lucene92HnswVectorsFormat.DEFAULT_MAX_CONN,
+                        Lucene92HnswVectorsFormat.DEFAULT_BEAM_WIDTH
+                    )
+                );
+                return new Lucene92HnswVectorsFormat();
+            }
+            final KNNVectorFieldMapper.KNNVectorFieldType type = (KNNVectorFieldMapper.KNNVectorFieldType) mapperService.get()
+                .fieldType(field);
+            final Map<String, Object> params = type.getKnnMethodContext().getMethodComponent().getParameters();
+            int maxConnections = getMaxConnections(params);
+            int beamWidth = getBeamWidth(params);
+            log.debug(
+                String.format(
+                    "Initialize KNN vector format for field [%s] with params [max_connections] = \"%d\" and [beam_width] = \"%d\"",
+                    field,
+                    maxConnections,
+                    beamWidth
+                )
+            );
+            final KnnVectorsFormat luceneHnswVectorsFormat = new Lucene92HnswVectorsFormat(maxConnections, beamWidth);
+            return luceneHnswVectorsFormat;
+        }
+
+        private boolean isNotKnnVectorFieldType(final String field) {
+            return mapperService.isPresent() && mapperService.get().fieldType(field) instanceof KNNVectorFieldMapper.KNNVectorFieldType;
```

Comment:
needs explicit parentheses

Reply:
ack, I'll actually revert the condition because it say "isNot"